### PR TITLE
Adding `pyproject.toml`

### PR DESCRIPTION
**Pull request by mfixstsci**
Addresses #584

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -9,21 +9,25 @@ version: 2
...
+
+conda:
+  environment: docs/rtd_environment.yaml
 
 # Build documentation in the docs/ directory with Sphinx
 sphinx:
-   configuration: docs/conf.py
+  builder: html
+  configuration: docs/conf.py
+  fail_on_warning: true
```
Comment (zacharyburnett) [inline .readthedocs.yaml:21]: sorry! I shouldn't have set this to `true`; it's causing the RTD build to fail
```suggestion
  fail_on_warning: false
```

**Review by zacharyburnett — APPROVED** [review]

**Comment by mfixstsci** [comment]
@hover2pi this PR is ready to merge. @zacharyburnett approved the changes and we are ready to go.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  'bandit',
+  'batman-package',
+  'bibtexparser',
+  'bokeh',
+  'boto3',
+  'corner',
+  'cython',
+  'docopt',
+  'docutils',
+  'flake8',
```
Comment (zacharyburnett) [inline pyproject.toml:38]: I don't think this is needed in the wheel dependencies (unless it is directly imported by the application at runtime). Are there other dependencies like this that can be removed?
```suggestion
```

**Review by hover2pi — APPROVED** [review]
Excellent!

**Pull request merged**